File: support_tools/swigwin-3.0.12/Examples/test-suite/csharp/aggregate_runme.cs
using System;
using aggregateNamespace;

public class runme {
    static void Main() {

      // Confirm that move() returns correct results under normal use
      int result = aggregate.move(aggregate.UP);
      if (result != aggregate.UP) throw new Exception("UP failed");

      result = aggregate.move(aggregate.DOWN);
      if (result != aggregate.DOWN) throw new Exception("DOWN failed");

      result = aggregate.move(aggregate.LEFT);
      if (result != aggregate.LEFT) throw new Exception("LEFT failed");

      result = aggregate.move(aggregate.RIGHT);
      if (result != aggregate.RIGHT) throw new Exception("RIGHT failed");

      // Confirm that move() raises an exception when the contract is violated
      try {
        aggregate.move(0);
        throw new Exception("0 test failed");
      }
      catch (ArgumentOutOfRangeException) {
      }
    }
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/aggregate_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'aggregate'

include Aggregate

# Confirm that move() returns correct results under normal use
result = move(UP)
raise RuntimeError unless (result == UP)

result = move(DOWN)
raise RuntimeError unless (result == DOWN)

result = move(LEFT)
raise RuntimeError unless (result == LEFT)

result = move(RIGHT)
raise RuntimeError unless (result == RIGHT)

# Confirm that it raises an exception when the contract is violated
failed = false
begin
  move(0)
rescue RuntimeError
  failed = true
end
raise RuntimeError unless failed

